### ISSUE TITLE
[Issue]:  completely random next up recommendations

### ISSUE THREAD

**Issue by ArchemedIan** [jellyfin/jellyfin#7906]
### Please describe your bug

literally just suggests random episodes of a few series, randomly. the same files are suggested properly in plex, emby, and kodi

what i did to produce bug:

add an affected show.

related evidence and social posts:

https://bowtie-ent.com/stuff/2022-06-12%2021-29-42.mkv

https://www.reddit.com/r/jellyfin/comments/vazk8b/comment/ic5k9wf/?utm_source=share&utm_medium=web2x&context=3

https://discord.com/channels/772232779534172171/773650260903591946/985704542836490280

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS: server: debian
- Virtualization: no
- Clients: any
- Browser: any
- FFmpeg Version: tried latest 4.x and 5.x
- Playback Method: n/a
- Hardware Acceleration: n/a
- Plugins: vanilla
- Reverse Proxy: local
- Base URL: local
- Networking: local 
- Storage: 18tb btrfs raid0
```

### Jellyfin logs

see attached

### Please attach any screenshots here

![image](https://user-images.githubusercontent.com/105951210/173265183-1974d21c-3425-4be7-86f5-9296a9c414c2.png)

[log.txt](https://github.com/jellyfin/jellyfin/files/8887305/log.txt)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cvium** [comment]
Do you have NFOs?

**Comment by kimpig** [comment]
I have the same problem. After downgrading to version 10.7.7, I checked and found that adding media after version 10.8.0 causes problems in Next Up.

**Comment by ArchemedIan** [comment]
> Do you have NFOs for the episodes?

I did, but I've made a test library with a different computer and copied the files over, and deleted the NFOs and srts recursively just to rule stuff out. I'm extremely frustrated with this lol. not at the devs or jellyfin really, btw, I'm frustrated that I cant find the source of the problem. every source of metadata I can think of that could be attached to the media, I've deleted. I sent them through multiple "clean metadata, subtitle, etc." plugins through tdarr, I've wiped ACLs because filebot puts stuff in them sometimes, I'm on a completely fresh debian install on another computer. ive tried setting metadata to specific providers and refreshing.

**Comment by cvium** [comment]
Can you send me your library.db database file?

**Comment by kimpig** [comment]
> Can you send me your library.db database file? I am unable to reproduce the issue myself.

Can I send it to you? If so, where can I send it?

**Comment by kimpig** [comment]
> If you are on Matrix you can PM me there claz:matrix.org or send it to this temp email [powesag435@serosin.com](mailto:powesag435@serosin.com)

I sent it by email.

**Comment by cvium** [comment]
And anyone can send me their library.db, but please tell me which series are exhibiting the problem

**Comment by kimpig** [comment]
> And anyone can send me their library.db, but please tell me which series are exhibiting the problem

Most of the recently added series have problems, but if I had to pick one, it would be 恋は世界征服のあとで.
PS) Should I tell you in the original title or in my local language?

**Comment by cvium** [comment]
It's stored in the database as  the language you've chosen... so probably local, but you can include both 😅

**Comment by kimpig** [comment]
> 

Okay. There is a problem with 
카구야 님은 고백받고 싶어 ~천재들의 연애 두뇌전~ (かぐや様は告らせたい～天才たちの恋愛頭脳戦～) 
and
사랑은 세계정복 후에 (恋は世界征服のあとで)

**Comment by cvium** [comment]
Is there a problem with RPG 부동산? Also btw, it's easier to do this over Matrix.

**Comment by kimpig** [comment]
> Is there a problem with RPG 부동산? Also btw, it's easier to do this over Matrix.

yes it has a problem too.

**Comment by cvium** [comment]
Something is setting `ForcedSortName`, which causes the order to be this for 사랑은 세계정복 후에
![image](https://user-images.githubusercontent.com/675118/174272984-69395e6f-6f4c-4307-849b-8fe50a55f7f4.png)

**Comment by kimpig** [comment]
This?
![image](https://user-images.githubusercontent.com/61234733/174274152-8f06cae3-4da6-45d8-9b5c-6ed9ae86bd63.png)

and I joined matrix.

**Comment by cvium** [comment]
> This?
> 
> ![image](https://user-images.githubusercontent.com/61234733/174274152-8f06cae3-4da6-45d8-9b5c-6ed9ae86bd63.png)


Yes

**Comment by ArchemedIan** [comment]
library.db emailed, it was too big for my email to send, so im hosting it on the webserver jellyfin is on. affected show is criminal minds

**Comment by ArchemedIan** [comment]
i think this is confirmed a takt issue for me, started a new account and I'm no longer experiencing the issue at all. closing

**Comment by ArchemedIan** [comment]
ok, I don't even have trakt installed now, where do I find my library.db on a normal debian install, @cvium

![no trakt](https://user-images.githubusercontent.com/105951210/177196857-177e3115-76ab-4845-8808-e75d792e3671.jpg)

**Comment by ArchemedIan** [comment]
ps, i literally woke up to this, was working fine for days

**Comment by ArchemedIan** [comment]
pps i ahve not done any downgrading, i updated to 10.8.1 from 10.8.0 when it came out, other than that, nothing was manually changed

**Comment by ArchemedIan** [comment]
found it https://bowtie-ent.com/stuff/JellyfinLibrary-07-04-22.db

**Comment by kimpig** [comment]
Do you have "Enable Rewatching in Next Up" turned on in your display settings?

In my case, the problem was the metadata generated by Jellyfin 10.8.0 beta.

**Comment by YourSandwich** [comment]
Since upgrade to 10.8.4 all new Series i added to jellyfin did not fetch metadata or show episodes. So i deleted the librery.db and rescaned everything back. The old issue is gone but now some random 1st epsiodes are diplayed in Next Up.
![image](https://user-images.githubusercontent.com/20018243/187235456-2442cb23-d16a-41f7-a94a-3901db8bcd3e.png)

Is this related to this Issue?

**Comment by ggogel** [comment]
TL;DR: I deleted all NFO files and re-created my library. This fixed the issue for most shows but it still appears for the show `Modern Family`.

Since I only described this in the mentioned reddit post, here a quick revise about my environment.

I run emby and jellyfin side-by-side. Jellyfin only had read access to the media folders. The metadata was created by emby and written to NFO files.

What I did:
1) Removed write access for emby on media folders
2) Deleted all NFO files belonging to episodes
3) Stopped jellyfin
4) Deleted library.db
5) Gave jellyfin write access to the media folders.
6) Ran a library scan
7) Imported all watch states with [watchstate](https://github.com/ArabCoders/watchstate)

It seems like the affected show doesn't have any `sorttitle` set, except for the `tvshow` object. I checked this by executing the following command in the show's folder:

```
grep -i -r -n --include \*.nfo sorttitle ./
```

Which gave the output:
```
./tvshow.nfo:12:  <sorttitle>Modern Family</sorttitle>
```

Also when I check `Edit Metadata` in the UI, the episodes don't have `sorttitle` set.

I remember that right after I re-created the library, Next Up was showing the correct episode. Then I watched some episodes in Emby. The watchstate tool synced the watchstate over to Jellyfin. I've also set up the Trakt plugin again in the meantime, which likely also synced some stuff. This leads me to believe that an incorrect `sorttitle` attribute might not be the only cause for this issue.

**Comment by cvium** [comment]
Possibly fixed by https://github.com/jellyfin/jellyfin/pull/8348

**Comment by blackwind** [comment]
@cvium Not fixed, or at least not entirely. I have one particular show that isn't listed on any of the metadata providers, doesn't have season or episode numbers, doesn't have `ForcedSortTitle`s, but does have chronological titles ("Episode 200: Having Some Fun", "Episode 201: This Is a Great Day"...). Choosing the next up episode by title or date added would be sufficient to produce the expected result, but I've investigated in `library.db`, and Jellyfin is definitely choosing the next episode based on `PresentationUniqueKey`. I have about 40 episodes of the show in question currently, I've run an SQL query to display only those episodes, I've sorted by `PresentationUniqueKey`, and each time I mark the next up episode as watched, Jellyfin always chooses the next `PresentationUniqueKey` in alphabetical order as its next episode.

**Comment by kimpig** [comment]
Hello, @cvium I checked recently, but it still hasn't been modified.

If I watch the entire season 1 of the TV show, it will be shown in Next Up from the last episode of season 2. However, if I watch Episode 1 of Season 2, Next Up will show Episode 2 of Season 2. I'm not sure what the problem is.

**Comment by crobibero** [comment]
Try this PR
https://github.com/jellyfin/jellyfin/pull/8662

**Comment by blackwind** [comment]
@crobibero It sounds like @kimpig's bug (which I also experienced) is resolved via #8662 and v10.8.7, but my bug persists. Should I file a new issue?
